app/src/main/java/com/rockthevote/grommet/ui/registration/RegistrationViewModel.kt

```diff
@@ -98,55 +99,62 @@ class RegistrationViewModel(
     ) {
         storeReviewData(data)
 
-        runCatching {
-            val transformer = RegistrationDataTransformer(currentRegistrationData, sessionData, completionDate)
-            val requestData = transformer.transform()
-
-            /* TODO Either inject an adapter, make this an async operation, or just pass the data
-                object to whatever class handles storing this to offload the responsibility
-             */
-            val adapter = Moshi.Builder()
-                .add(KotlinJsonAdapterFactory())
-                .build()
-                .adapter(RockyRequest::class.java)
-
-            val rockyRequestJson = adapter.toJson(requestData)
-
-            val registrantName = with(currentRegistrationData.newRegistrantData!!.name) {
-                listOfNotNull(firstName, middleName, lastName).joinToString(" ")
-            }
-
-            val registration = Registration(
-                registrationDate = completionDate.time,
-                registrantName = registrantName,
-                // This should only used if there was already an error on registration
-                //  so allowing an empty email is acceptable in this case
-                registrantEmail = currentRegistrationData.additionalInfoData?.emailAddress ?: "",
-                registrationData = rockyRequestJson
-            )
+        viewModelScope.launch(dispatcherProvider.io) {
+
+            val currentSession = sessionDao.getCurrentSession()
+
+            val sessionData = currentSession?.let {
+                SessionData(
+                    partnerId = currentSession.partnerInfoId,
+                    canvasserName = currentSession.canvasserName,
+                    sourceTrackingId = currentSession.sourceTrackingId,
+                    partnerTrackingId = currentSession.partnerTrackingId,
+                    geoLocation = GeoLocation(
+                        lat = currentSession.geoLocation.latitude(),
+                        long = currentSession.geoLocation.longitude()
+                    ),
+                    openTrackingId = currentSession.openTrackingId
+                )
+            } ?: SessionData(
+                partnerId = -1,
+                canvasserName = "empty",
+                sourceTrackingId = "empty",
+                partnerTrackingId = "empty",
+                geoLocation = GeoLocation(-1.0, -1.0),
+                openTrackingId = "empty"
+            ) // TODO this is default if a session doesn't exist, probably good to keep it just in case?
```

Comment:
I would think we'd rather throw an exception if we get here, the thought being they correct the issue and the person re-registers.

If this scenario happens, then there's probably going to be a lot more than just one registration w/o session data which will be a PITA to explain.

We're going to have a check to make sure there's an active session before they can register anyone, so we will have a hard check against this happening. If they do get here, and we do throw an exception or warn them and stop them from registering, then they could potentially get into a situation where they can't fix it anyway.
ex:

1. warn, don't allow to proceed
1. they have to abandon the registration and go back to enter session info
1. before they can do that they have to clock out, before that they have to upload all registrations
1. registrations fail because no session info

So, on second thought, i think we should keep the default you have and just make sure log it to firebase once we get to that part. A timber log is fine for now.

Reply:
created this to track https://mechanical-man.atlassian.net/browse/GROM-145

Reply:
Gotcha, yeah. This is the way it is now because `sessionDao.getCurrentSession()` literally returns nothing now because we're not inputting sessions into the DB. Generally, I'd treat this as a true exception, but it seems like the generally way of handling errors on the project is a soft fail on server side (eg. 200 with the expectation of remediation on the backend if possible). Right now this is required, but should theoretically never happen in production once we're inputting sessions into the DB.

I'll add a log statement, but have no qualms about either throwing an exception _or_ continuing to use the default in the event of an error.